.ruff.toml

```diff
@@ -0,0 +1,309 @@
+    "PT022",   # pytest-useless-yield-fixture
+    "PT023",   # pytest-incorrect-mark-parentheses-style
+
+    # flake8-use-pathlib (PTH)
+    "PTH100",  # os-path-abspath
+    "PTH101",  # os-chmod
+    "PTH102",  # os-mkdir
+    "PTH103",  # os-makedirs
+    "PTH104",  # os-rename
+    "PTH107",  # os-remove
+    "PTH108",  # os-unlink
+    "PTH109",  # os-getcwd
+    "PTH110",  # os-path-exists
+    "PTH111",  # os-path-expanduser
+    "PTH112",  # os-path-isdir
+    "PTH113",  # os-path-isfile
+    "PTH114",  # os-path-islink
+    "PTH116",  # os-stat
+    "PTH118",  # os-path-join
+    "PTH119",  # os-path-basename
+    "PTH120",  # os-path-dirname
+    "PTH122",  # os-path-splitext
+    "PTH123",  # builtin-open
+    "PTH124",  # py-path
+
+    # flake8-return (RET)
+    "RET501",  # unnecessary-return-none
+    "RET502",  # implicit-return-value
+    "RET503",  # implicit-return
+    "RET504",  # unnecessary-assign
+    "RET505",  # superfluous-else-return
+    "RET506",  # superfluous-else-raise
+    "RET507",  # superfluous-else-continue
+    "RET508",  # superfluous-else-break
+
+    # flake8-raise (RSE)
+    "RSE102",  # unnecessary-paren-on-raise-exception
+
+    # Ruff-specific rules (RUF)
+    "RUF001",  # ambiguous-unicode-character-string
+    "RUF002",  # ambiguous-unicode-character-docstring
+
+    # flake8-bandit (S)
+    "S101",  # Use of `assert` detected
+    "S105",  # hardcoded-password-string
+    "S110",  # try-except-pass
+    "S112",  # try-except-continue
+    "S301",  # suspicious-pickle-usage
+    "S311",  # suspecious-non-cryptographic-randomness
+    "S324",  # hashlib-insecure-hash-function
+    "S506",  # UnsafeYAMLLoad
+    "S310",  # Suspicious-url-open-usage
+    "S321",  # Suspecious-ftp-lib-usage
+    "S603",  # `subprocess` call: check for execution of untrusted input
+    "S605",  # Starting a process with a shell, possible injection detected  (2023-05-03)
+    "S607",  # Starting a process with a partial executable path
+    "S608",  # Posslibe SQL injection
+
+    # flake8-simplify (SIM)
+    "SIM102",  # NestedIfStatements
+    "SIM105",  # UseContextlibSuppress
+    "SIM108",  # UseTernaryOperator
+    "SIM114",  # if-with-same-arms
+    "SIM115",  # OpenFileWithContextHandler
+    "SIM117",  # MultipleWithStatements
+    "SIM118",  # KeyInDict
+    "SIM201",  # NegateEqualOp
+    "SIM202",  # NegateNotEqualOp
+    "SIM300",  # yoda condition
+
+    # flake8-print (T20)
+    "T201",  # PrintUsed
+
+    # tryceratops (TRY)
+    "TRY002",  # raise-vanilla-class
+    "TRY003",  # raise-vanilla-args
+    "TRY004",  # prefer-type-error
+    "TRY200",  # reraise-no-cause
+    "TRY201",  # verbose-raise
+    "TRY300",  # Consider `else` block
+    "TRY301",  # raise-within-try
+    "TRY400",  # error-instead-of-exception
+
+    # flake8-quotes (Q)
+    "Q000",  # use double quotes
+]
+unfixable = [
+    "E711"  # NoneComparison. Hard to fix b/c numpy has it's own None.
+]
```

Comment:
Does this belong here or should it be in `pyproject.toml`?

Reply:
I would like to fix this eventually.
Or ask ruff if this can be detected with a sufficiently low false-positive rate.